### PR TITLE
LibWeb: Support SRI for import maps

### PR DESCRIPTION
**Pull request by jamierocks**
I wrote this a while ago, and never made a PR - not entirely sure why, but better late than never :^)

See:
- https://github.com/whatwg/html/commit/b2fdca1
- http://wpt.live/import-maps/static-integrity.html

### CODE REVIEW

File: Userland/Libraries/LibWeb/Bindings/MainThreadVM.cpp [hunk 1]
```diff
@@ -422,8 +422,8 @@ ErrorOr<void> initialize_main_thread_vm(HTML::EventLoop::Type type)
         // 3. Let referencingScript be null.
         Optional<HTML::Script&> referencing_script;
 
-        // FIXME: 4. Let fetchOptions be the default classic script fetch options.
-        auto fetch_options = HTML::default_classic_script_fetch_options();
+        // FIXME: 4. Let originalFetchOptions be the default classic script fetch options.
```
Comment (tcl3) [inline Userland/Libraries/LibWeb/Bindings/MainThreadVM.cpp:425]: Is the FIXME still needed here?

File: Userland/Libraries/LibWeb/HTML/Scripting/Fetching.cpp [hunk 2]
```diff
@@ -280,6 +280,50 @@ static void set_up_module_script_request(Fetch::Infrastructure::Request& request
     request.set_priority(options.fetch_priority);
 }
 
+// https://html.spec.whatwg.org/multipage/webappapis.html#get-the-descendant-script-fetch-options
+WebIDL::ExceptionOr<ScriptFetchOptions> get_descendant_script_fetch_options(ScriptFetchOptions original_options, const URL::URL& url, EnvironmentSettingsObject& settings_object)
```
Comment (tcl3) [inline Userland/Libraries/LibWeb/HTML/Scripting/Fetching.cpp:284]: East const should be used for `url`.

Do we need to create pass `original_options` by value, or can we use a const reference

File: Userland/Libraries/LibWeb/HTML/Scripting/ImportMap.cpp [hunk 3]
```diff
@@ -209,4 +225,47 @@ WebIDL::ExceptionOr<HashMap<URL::URL, ModuleSpecifierMap>> sort_and_normalise_sc
     return normalised;
 }
 
+// https://html.spec.whatwg.org/multipage/webappapis.html#normalizing-a-module-integrity-map
+WebIDL::ExceptionOr<ModuleIntegrityMap> normalise_module_integrity_map(JS::Realm& realm, JS::Object& original_map, URL::URL base_url)
```
Comment (tcl3) [inline Userland/Libraries/LibWeb/HTML/Scripting/ImportMap.cpp:229]: The same spelling as is used in the specification should be used for "normalize".

File: Userland/Libraries/LibWeb/HTML/Scripting/Fetching.cpp [hunk 4]
```diff
@@ -280,6 +280,50 @@ static void set_up_module_script_request(Fetch::Infrastructure::Request& request
...
+{
+    // 1. Let newOptions be a copy of originalOptions.
+    ScriptFetchOptions new_options;
+    new_options.cryptographic_nonce = original_options.cryptographic_nonce;
+    new_options.integrity_metadata = original_options.integrity_metadata;
+    new_options.parser_metadata = original_options.parser_metadata;
+    new_options.credentials_mode = original_options.credentials_mode;
+    new_options.referrer_policy = original_options.referrer_policy;
+    new_options.render_blocking = original_options.render_blocking;
+    new_options.fetch_priority = original_options.fetch_priority;
```
Comment (tcl3) [inline Userland/Libraries/LibWeb/HTML/Scripting/Fetching.cpp:294]: Do we need to copy each member like this or would something like `auto new_options = original_options` work?

**Review by tcl3 — COMMENTED** [review]
Is there any way that we can test this feature?

File: Userland/Libraries/LibWeb/Bindings/MainThreadVM.cpp (hunk 1 above)
Comment (jamierocks) [reply Userland/Libraries/LibWeb/Bindings/MainThreadVM.cpp:425]: I wondered the same thing earlier - I couldn't work out why it was still in place. I'll remove it :)

**Review by tcl3 — APPROVED** [review]

**Pull request merged**